app/reducers/Session.js

```diff
@@ -0,0 +1,152 @@
+import Immutable from 'immutable';
+
+import { NEW_SESSION_REQUESTED, NEW_SESSION_BEGAN, NEW_SESSION_DONE,
+        SAVE_SESSION_REQUESTED, SAVE_SESSION_DONE, GET_SAVED_SESSIONS_REQUESTED, GET_SAVED_SESSIONS_DONE,
+        SESSION_LOADING, SESSION_LOADING_DONE,
+        SET_CAPABILITY_PARAM, ADD_CAPABILITY, REMOVE_CAPABILITY, SET_CAPS,
+        SWITCHED_TABS, SAVE_AS_MODAL_REQUESTED, HIDE_SAVE_AS_MODAL_REQUESTED, SET_SAVE_AS_TEXT,
+        DELETE_SAVED_SESSION_REQUESTED, DELETE_SAVED_SESSION_DONE, 
+        CHANGE_SERVER_TYPE, SET_SERVER_PARAM, SET_SERVER,
+        ServerTypes } from '../actions/Session';
+
+// Make sure there's always at least one cap
+const INITIAL_STATE = {
+  savedSessions: [],
+  tabKey: 'new',
+  serverType: ServerTypes.local,
+  server: {
+    local: {},
+    remote: {},
+    sauce: {},
+  },
+  caps: [{
+    type: 'text',
+  }],
+  isCapsDirty: true,
+};
+
+export default function session (state=INITIAL_STATE, action) {
+  switch (action.type) {
+    case NEW_SESSION_REQUESTED:
+      return Immutable.fromJS(state)
```

Comment:
My original motivations

1) **Cleaner Code** The ES6 conventions worked great for shallow copying but when deep copying was required the code became messy. I also like how Immutable operations sententially describe how the state is being changed.

e.g.)

```javascript
// With immutable.js
case SET_CAPABILITY_PARAM:
  return Immutable.fromJS(state)
    .setIn(['caps', action.index, action.name], action.value)
    .set('isCapsDirty', true)
    .toJS();
```

```javascript
// Without immutable
case SET_CAPABILITY_PARAM:
  let caps = {...nextState.caps};
  caps[action.index][action.name] = action.value;
  return {
    ...state,
    ...caps,
    isCapsDirty: true,
  }
```

2) **Enforced immutability** Using the standard ES6 convention I had mistakenly forgotten to deep copy elements which broke the React components (React expects a completely new state). I was able to spot them and fix them but it took some time and there's a risk that myself or another dev may make this mistake in the future without realizing it. ImmutableJS eliminates this class of risks altogether.

On second thought though...

I think it may be better to make use of 'combineReducers' (http://redux.js.org/docs/api/combineReducers.html) so that all data is shallow